packages/webapp/src/containers/Finances/AddSale/RevenueTypes/useSortedRevenueTypes.jsx

```diff
@@ -0,0 +1,63 @@
+/*
+ *  Copyright 2023 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+import { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import PropTypes from 'prop-types';
+import { useTranslation } from 'react-i18next';
+import { getRevenueTypes } from '../../saga';
+import { revenueTypesSelector } from '../../../revenueTypeSlice';
+
+export default function useSortedRevenueTypes() {
+  const dispatch = useDispatch();
+  const { t } = useTranslation();
+
+  const [sortedtypes, setSortedTypes] = useState([]);
+  const revenueTypes = useSelector(revenueTypesSelector);
+
+  useEffect(() => {
+    dispatch(getRevenueTypes());
+  }, []);
+
+  useEffect(() => {
+    const defaultTypes = [];
+    const customTypes = [];
+    revenueTypes?.forEach((type) => {
+      if (!type.deleted) {
```

Comment:
I don't mind an extra check at all but that particular selector does already filter for deleted, so it's not... possible, right? That `type.deleted` could be true?

And ooops in checking this I just realized I have a useless destructuring of `deleted` in that selector that I didn't notice before 😬

Reply:
Thank you Joyce!

No, you are right, I used a selector that selects all types including deleted ones for expense type, and I forgot to remove the check from the code I copied... I will remove it!
I spotted the unused `deleted` but forgot to remove. Shall I remove it? If you made a commit already, I will leave it!

Reply:
Would you remove it for me?? 😍 No I didn't make a commit!

Reply:
Done 😄